src/DraggableTreeviewNode.vue

```diff
@@ -125,12 +128,30 @@ export default Vue.extend({
   },
   watch: {
     value(value): void {
-      this.localValue = { ...value };
+      this.localValue = {
+        ...value,
+        children: [...this.value.children],
+      };
     },
   },
   methods: {
-    updateValue(value: TreeItem[]): void {
-      this.localValue.children = [...value];
+    updateValue($event: any): void {
+      // removing must be done before adding
+      // in order for the event `moved` to work properly
+      const dataForRemoving = $event.removed || $event.moved;
+      if (dataForRemoving) {
+        this.localValue.children.splice(dataForRemoving.oldIndex, 1);
+      }
+
+      const dataForAdding = $event.added || $event.moved;
+      if (dataForAdding) {
+        this.localValue.children.splice(
+          dataForAdding.newIndex,
+          0,
+          dataForAdding.element
+        );
+      }
+
```

Comment:
the mutating error is fixed by the 2 other changes that create a shallow copy of children for `localValue`

but without this change, a strange bug appears : 

https://github.com/suusan2go/vuetify-draggable-treeview/assets/16286023/ba4e0725-8135-4023-9921-e0eb555409d8

on drag/drop, multiple updates occur : 
1.  the child "test 2" gets updated with its now 4 children
2. the parent "Plats cuisinés" gets updated with its now only 1 child "test 2" (but the value passed of "test 2" has only the 3 original children, not the now 4)